core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java

```diff
@@ -573,20 +576,21 @@ private void forceClose() {
   }
 
   private boolean isAuthFailure(Throwable error) {
-    boolean authFailure = true;
     if (error instanceof AllNodesFailedException) {
-      Collection<Throwable> errors = ((AllNodesFailedException) error).getErrors().values();
+      Collection<List<Throwable>> errors =
+          ((AllNodesFailedException) error).getAllErrors().values();
       if (errors.size() == 0) {
         return false;
       }
-      for (Throwable nodeError : errors) {
-        if (!(nodeError instanceof AuthenticationException)) {
-          authFailure = false;
-          break;
+      for (List<Throwable> nodeErrors : errors) {
+        for (Throwable nodeError : nodeErrors) {
+          if (!(nodeError instanceof AuthenticationException)) {
```

Comment:
is this refactoring correct? Previously if any of the errors were `AuthenticationException` then this method returned `true`. Now if any of the errors is NOT `AuthenticationException` then this method returns `false`.

Reply:
I don't agree, both algorithms look equivalent to me and both return `false` as soon as any error is NOT `AuthenticationException`.

Reply:
ah, ok: there is a `break` there, thx for the clarification